frontend/src/components/CourseCard/CourseCard.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import QuickCart from 'components/QuickCart';
+import { addTab } from 'reducers/courseTabsSlice';
+import S from './styles';
+
+type Props = {
+  courseCode: string
+  title: string
+  planned: boolean
+};
+
+const CourseCard = ({ courseCode, title, planned }: Props) => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  const handleCourseLink = (e: React.MouseEvent<HTMLElement>): void => {
+    e.stopPropagation();
+    navigate('/course-selector');
+    dispatch(addTab(courseCode));
+  };
+
+  return (
+    <S.CourseCard
+      planned={planned}
+      type="primary"
+      onClick={handleCourseLink}
+    >
+      <S.CourseWrapper>
+        <S.CourseCode>{courseCode}</S.CourseCode>
+        <S.CourseTitle>{title}</S.CourseTitle>
+      </S.CourseWrapper>
+      <S.QuickCartButton>
```

Comment:
what is the point in having a flexbox wrapper for the quick cart button? Let's also rename this to something like `S.QuickCartWrapper` instead if you intend to keep it.